File: codeforces/Gym/Esquenta FACOMpetindo/f.py
import bisect

t = int(input())
    
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    
    pre = [0] * n
    pre[0] = a[0]
    for i in range(1, n):
        pre[i] = pre[i-1] + a[i]
    
    ans = n
    
    for i in range(n):
        soma_segmento = pre[i]
        soma_total = soma_segmento * 2
        
        curr = i + 1
        last = i
        
        while soma_total in pre:
            idx = bisect.bisect_left(pre, soma_total)
            soma_total += soma_segmento
            curr = max(curr, idx-last)
            last = idx

            if last == n - 1:
            	ans = min(ans, curr)
    
    print(ans)

File: codeforces/Gym/Esquenta FACOMpetindo/f.cpp
#include <iostream>
#include <algorithm>
 
using namespace std;
 
int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr);
 
    int t;
    cin >> t;
 
    while (t--) {
        int n;
        cin >> n;
 
        int a[n];
 
        for (int i = 0; i < n; i++) 
            cin >> a[i];
 
        int pre[n];
        pre[0] = a[0];
 
        for (int i = 1; i < n; i++)
            pre[i] = pre[i-1] + a[i];
 
        int ans = n;
 
        for (int i = 0; i < n; i++) {
            int sum = pre[i];
            sum += sum;
 
            int curr = (i+1), last = i;
            int idx;
 
            while (binary_search(pre, pre+n, sum)) {
                idx = lower_bound(pre, pre+n, sum)-pre;
                sum += pre[i];
                curr = max(curr, idx-last);
                last = idx;
            
                if (last == (n-1))
                    ans = min(ans, curr);
            }
        }
 
        cout << ans << "\n";
    }
}